arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/ReturnSection.java

```diff
@@ -290,7 +291,8 @@ public void resolveUponRollback(Hub hub, MessageFrame messageFrame, CallFrame ca
             ? RETURN_FROM_DEPLOYMENT_NONEMPTY_CODE_WILL_REVERT
             : RETURN_FROM_DEPLOYMENT_EMPTY_CODE_WILL_REVERT);
 
-    undoingDeploymentAccountSnapshot = AccountSnapshot.canonical(hub, deploymentAddress);
+    secondCreatee = firstCreateeNew.deepCopy().setDeploymentNumber(hub);
+    secondCreateeNew = firstCreatee.deepCopy().setDeploymentNumber(hub);
```

Comment:
No more "taking snapshots at revert time". This was the root cause for shit going sideways all the time.